src/main/java/no/unit/nva/cristin/projects/CristinHandler.java

```diff
@@ -30,4 +43,18 @@ protected static String getValidLanguage(RequestInfo requestInfo) throws BadRequ
         return attempt(() -> requestInfo.getQueryParameter(queryParameter)).toOptional();
     }
 
+    protected void addRequestedContentTypeToResponseIfSupported(RequestInfo requestInfo) {
+        getRequestedContentType(requestInfo)
+            .map(String::toLowerCase)
+            .filter(VALID_CONTENT_TYPES::contains)
+            .ifPresent(this::addContentTypeHeader);
+    }
+
+    private void addContentTypeHeader(String contentType) {
+        this.additionalHeaders.put(HttpHeaders.CONTENT_TYPE, contentType);
+    }
+
+    private Optional<String> getRequestedContentType(RequestInfo requestInfo) {
+        return Optional.ofNullable(requestInfo.getHeaders().get(HttpHeaders.ACCEPT));
```

Comment:
Optional.ofNullable(requestInfo).map(r->r.getHeaders()).map(h->h.get(HttpHeaders.Accept)). Better to chain the maps in order to avoid null pointer exceptions.

Reply:
The '-> ' that I used was only for brevity. Use proper style and variable names.